src/utils/transactions.ts

```diff
@@ -18,18 +18,21 @@ along with the library. If not, see <http://www.gnu.org/licenses/>.
 
 import { Transaction } from '@alephium/sdk/api/explorer'
 
+import { AddressPendingTransaction, AddressTransaction, PendingTransaction } from '../types/transactions'
+
 export const getNewTransactions = (
   incomingTransactions: Transaction[],
   existingTransactions: Transaction[]
 ): Transaction[] =>
-  incomingTransactions.reduce((newTxs: Transaction[], incomingTx: Transaction) => {
-    const targetTxIndex = existingTransactions.findIndex((existingTx) => existingTx.hash === incomingTx.hash)
-    if (targetTxIndex === -1) {
-      newTxs.push(incomingTx)
-    } else {
-      if (existingTransactions[targetTxIndex].blockHash === '') {
-        existingTransactions.splice(targetTxIndex, 1, incomingTx)
-      }
-    }
-    return newTxs
-  }, [])
+  incomingTransactions.filter((newTx) => !existingTransactions.some((existingTx) => existingTx.hash === newTx.hash))
+
+export const getRemainingPendingTransactions = (
```

Comment:
Same comment as before, `extractRemainingPendingTransactions` to avoid confusion with endpoint call?

Reply:
haha, already done!

Reply:
Ah you did it already :D